migtests/scripts/run-schema-migration.sh

```diff
@@ -105,10 +105,18 @@ main() {
 	import_schema --continue-on-error t
 	run_ysql ${TARGET_DB_NAME} "\dt"
 
+	# Extract the major version of YugabyteDB from the version string
+	target_major_version=$(run_ysql "${TARGET_DB_NAME}" "SELECT version();" | grep -oE 'YB-([0-9]+\.[0-9]+)' | cut -d '-' -f2)
+	echo "Major YugabyteDB Version: $target_major_version"
+
     if [ -f "${EXPORT_DIR}/schema/failed.sql" ]
     then
+		EXPECTED_FAILED_FILE="${TEST_DIR}/expected_files/expected_failed_${target_major_version}.sql"
+		if [ ! -f "$EXPECTED_FAILED_FILE" ]; then
```

Comment:
While this is nice, one problem is that we'll have to go and add new files for all newer versions, whenever they come out. We should figure out a way to configure < 2.25 - file1, >=2.25 - file2. 

Let me think about this and get back.

Reply:
Created a separate ticket to track the extension from 2.25 to other versions like 2025.1 and beyond.
https://yugabyte.atlassian.net/browse/DB-16221